app/ai/voice/agents/breeze_buddy/managers/calls.py

```
@@ -281,6 +281,50 @@ async def _release_number(number_id: str, provider: CallProvider):
+    try:
+        redis = await get_redis_service()
+        already_released = not await redis.set(
+            guard_key, "1", nx=True, ex=_CHANNEL_RELEASE_GUARD_TTL
+        )
+        if already_released:
+            logger.info(
+                f"Channel already released for call_id: {call_id} "
+                f"(outbound_number: {number_id}), skipping duplicate decrement"
+            )
+            return
+    except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)